components/typography/Heading.tsx

```diff
@@ -0,0 +1,77 @@
+interface HeadingProps {
+  typeStyle?: 
+    | 'heading-xl'
+    | 'heading-lg'
+    | 'heading-md'
+    | 'heading-md-semibold'
+    | 'heading-sm'
+    | 'heading-sm-semibold'
+    | 'heading-xs'
+    | 'heading-xs-semibold'
+    | 'body-lg'
+    | 'body-md'
+    | 'body-sm';
```

Comment:
instead of giving the values in string, you should use the `enum` property inside typescript.